### DISCUSSION TITLE
TShock and/or Terraria Server crash on world load

### DISCUSSION THREAD

**Discussion by MrConorAE** [Pryaxis/TShock#2156] [Q&A] [answered]
Hello Terrarians!

I've just installed TShock (lastest pre-release). I installed it by getting the last full release, and cumulatively applying all the pre-releases on top of it, because some of the updates don't include others.

I've generated a world from the command-line and played on it fine. However, when trying to load a world that was previously played on with Steam Terraria (after copying it to .local/Terraria/Worlds), TShock and/or terrariaserver.exe crash (I'm not sure which is causing the crash). Here's the log:
```
Server crash: 11/14/2020 10:24:34 AM
System.NullReferenceException: Object reference not set to an instance of an object.
  at Terraria.Main.WriteFancyWorldLoadErrorToConsole () [0x00005] in <b823b82d9daf48e59d7ebe55aa874ffb>:0
  at Terraria.Main.DedServ () [0x00adf] in <b823b82d9daf48e59d7ebe55aa874ffb>:0
  at Terraria.Program.LaunchGame (System.String[] args, System.Boolean monoArgs) [0x000ac] in <b823b82d9daf48e59d7ebe55aa874ffb>:0

Please send crashlog.txt to support@terraria.org
```

Does anybody have any idea what's going on here, and how I could fix it? A quick web search only gives results about C# programming, nothing specific to Terraria.

Thank you so much!

**Comment by Kojirremer** [comment, marked as answer]
```installed it by getting the last full release, and cumulatively applying all the pre-releases on top of it, because some of the updates don't include others.```
What does this mean? Did you overwrite the latest update with files from previous versions? Updates should be 100% standalone.

**Reply by MrConorAE** [reply]
Here's how I did it:
```
- Download, unzip, install last stable version.
- For each pre-release since last stable:
  - Download, unzip, install pre-release, overwriting any files that are duplicate.
```
I'm still not too sure why I did this, but my vague idea was that since they were pre-releases, I should apply them on top of the stable. Also, some prereleases did not include files (e.g. most recent didn't have TerrariaServer.exe), so I thought they needed the previous versions to "build on".

**Reply by MrConorAE** [reply]
Do you think this might have caused it?

**Reply by Quinci135** [reply]
World is corrupted, world is a higher version than the server (make sure you are on the latest pre-release), an invalid world is not present in the given commandline argument (like you have a startup .bat or .sh that has a pre-defined world path)
one of those

**Reply by Kojirremer** [reply]
```world is a higher version than the server```
If you did overwrite the latest prerelease with files from previous ones, this is most likely what the error is pointing out.

**Reply by MrConorAE** [reply]
Hmmm....
I'm not using preset paths; the world is selected with the interactive prompt on startup.
The world is from latest Steam version (1.4.1.2). TShock... uploaded a pre-release for 1.4.1.2 13 hours ago (🤦 well done me). I'll install it and see if it helps. 
Thank you so much for putting up with my stupid mistakes :D

**Reply by MrConorAE** [reply]
Well, that did it! Sorry for bothering you guys, and thank you so much for your help.
